walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -19,40 +19,60 @@
 import java.util.List;
 
 public final class FindMeetingQuery {
+
+  private static final int MINUTES_IN_ONE_DAY = 1440;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     long duration = request.getDuration();
-    int[] minutes = new int[1441];
+    int[] minutes = new int[MINUTES_IN_ONE_DAY];
     List<TimeRange> res = new ArrayList<>();
     List<String> requestAttendees = new ArrayList(request.getAttendees());
+    List<String> optionalAttendees = new ArrayList(request.getOptionalAttendees());
 
     for (Event event : events) {
       int rangeStart = event.getWhen().start();
       int rangeEnd = event.getWhen().end();
 
       List<String> eventAttendees = new ArrayList(event.getAttendees());
 
-      int mustFill = 0;
+      boolean mustFill = false;
+      boolean optionalFill = false;
 
       for (String attendee : eventAttendees) {
         if (requestAttendees.contains(attendee)) {
-          mustFill = 1;
-          break;
+          mustFill = true;
+        }
+        if (optionalAttendees.contains(attendee)) {
+          optionalFill = true;
         }
       }
-      if (mustFill == 1) {
+
+      if (mustFill == true) {
         for (int i = rangeStart; i < rangeEnd; i++) {
           minutes[i] = 1;
         }
       }
+
+      if (mustFill == false && optionalFill == true) {
+        // If there is any previous mustFill I'm not changing.
+        // Otherwise I'm labeling with 2 because it's not preffered time.
+        // Because there is optional attendee.
+        for (int i = rangeStart; i < rangeEnd; i++) {
+          if (minutes[i] != 1) {
+            minutes[i] = 2;
+          }
+        }
+      }
     }
 
-    for (int i = 0; i < 1440; i++) {
+    // First try to add request optional attendees.
+    for (int i = 0; i < MINUTES_IN_ONE_DAY; i++) {
```

Comment:
You're using a very similar code twice. Please consider extracting it into a helper function.

Reply:
Thanks, I merged them with one function and two helper functions.